File: src/main/java/ru/sicampus/bootcamp2026/repositories/UserRepository.java
package ru.sicampus.bootcamp2026.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import ru.sicampus.bootcamp2026.entities.User;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);
    boolean existsByEmail(String email);
}

File: src/main/kotlin/ru/sicampus/bootcamp2026/repository/UserRepository.kt
package ru.sicampus.bootcamp2026.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.sicampus.bootcamp2026.entity.User
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): Optional<User>
}
